web-app/helpers/certificate-designs.ts

```diff
@@ -28,29 +27,105 @@ function addText(canvas: Canvas, text: string, font: string, fillStyle: string,
+    const testWidth = metrics.width;
+
+    // doesn't really understand the condition
+    // ask Ryan about it
+    if (testWidth > maxWidth && n > 6) {
+      context.fillStyle = fillStyle;
+      context.font = font;
+      // The x-axis coordinate of the point at which to begin drawing the text, in pixels.
+      // The y-axis coordinate of the baseline on which to begin drawing the text, in pixels.
+      context.fillText(line, x, y2);
+      line = `${words[n]}`;
+
+      y2 += lineHeight;
+    } else {
+      line = testLine.trim();
+    }
+  }
+  context.fillText(line, x, y2);
+}
+
+// eslint-disable-next-line max-lines-per-function
 export async function populateDeveloperCert(canvas: Canvas, details: any) {
-  // TODO
   console.log('populateDeveloperCert', { details });
-  const { tokenId, date, programName, accountName, expiration } = details;
+  const { tokenId, date, programName, accountName, expiration, programDescription, instructor, programCode } = details;
 
-  const fillStyle = `#${Math.floor(Math.random() * 16777215).toString(16)}`; // This is just temporary and will be removed once we have final designs for each program's certificate. https://css-tricks.com/snippets/javascript/random-hex-color/
-  const font = `40px '${fontFamily}' bold`;
+  const gray = '#757575';
+  const black = '#000000';
+  const blue = '#5F8AFA';
+  // font weight doesn't really work at all!
+  const accountFont = `60px '${fontFamily2}' medium`;
+  const dateFont = `30px '${fontFamily2}' medium`;
+  const descriptionFont = `33px '${fontFamily}' regular`;
+  const tokenIdFont = `30px '${fontFamily2}' medium`;
+  const programFont = `40px '${fontFamily2}' medium`;
+  const titleFont = `64px '${fontFamily}' extraBold`;
 
   // Load and draw the background image first
-  const certificateBackgroundSvgImage = './public/certificate-backgrounds/NCD_background.svg'; // Background images must be in SVG format
-  const image = await loadImage(certificateBackgroundSvgImage);
-  console.log({ image });
+  // Background images must be in SVG format
+  const certificateBackgroundNcdImage = './public/certificate-backgrounds/NCD_certificate.svg';
+  const certificateBackgroundNcaImage = './public/certificate-backgrounds/NCA_certificate.svg';
+  const certificateBackgroundNceImage = './public/certificate-backgrounds/NCE_certificate.svg';
+  const certificateBackgroundNciImage = './public/certificate-backgrounds/NCI_certificate.svg';
+  const certificateBackgroundNcarImage = './public/certificate-backgrounds/NCAR_certificate.svg';
+  const certificateBackgroundNcsImage = './public/certificate-backgrounds/NCS_certificate.svg';
+  const certificateBackgroundNcuxImage = './public/certificate-backgrounds/NCUX_certificate.svg';
+
+  let image;
+
+  switch (programCode) {
+    case 'TR101': // programCode needs to change in a meaningful way.
+      image = await loadImage(certificateBackgroundNcdImage);
+      console.log({ image });
```

Comment:
You have `console.log({ image });` repeated many times in this function. Please move it below the switch if it achieves what you're trying to do.